ytypes/unmarshal.go

```diff
@@ -29,6 +30,49 @@ type UnmarshalOpt interface {
 	IsUnmarshalOpt()
 }
 
+// ComplianceErrors contains the compliance errors encountered from an Unmarshal operation.
+type ComplianceErrors struct {
+	// Errors represent generic errors for now, until we make a decision on what specific types
+	// of errors should be returned.
+	Errors []error
+}
+
+func (c *ComplianceErrors) Error() string {
+	if c == nil {
+		return ""
+	}
+
+	var b strings.Builder
+	b.WriteString("Noncompliance errors:")
+	if len(c.Errors) != 0 {
+		for _, e := range c.Errors {
+			b.WriteString("\n\t")
+			b.WriteString(e.Error())
+		}
+	} else {
+		b.WriteString(" None")
+	}
+	b.WriteString("\n")
+	return b.String()
+}
+
+func (c *ComplianceErrors) append(errs... error) *ComplianceErrors {
+	if c == nil {
+		return &ComplianceErrors{Errors: errs}
+	}
+
+	c.Errors = append(c.Errors, errs...)
+	return c
+}
+
+// BestEffortUnmarshal is an unmarshal option that accumulates errors while unmarshalling,
+// and continues the unmarshaling process. An unmarshal now return a ComplianceErrors struct,
+// instead of a single error.
+type bestEffortUnmarshal struct{}
```

Comment:
Since we have decided on an MVP, if you intend to use this prior to adding more functionality then feel free to make this exported.

Reply:
Ack.